backend/file-orchestrator-service/util/src/main/kotlin/FileCollectionService.kt

```diff
@@ -56,6 +56,54 @@ class FileCollectionService(
         session: AsyncDBConnection,
         allowDuplicates: Boolean
     ) {
+        //Change to not allow drives to have same name within same provider
+        val error = session.sendPreparedStatement(
+            {
+                setParameter("isproject", !actorAndProject.project.isNullOrBlank())
+                setParameter("project", actorAndProject.project)
+                setParameter("username", actorAndProject.actor.safeUsername())
+            },
+            """
+                select title, r.provider, pc.category
+                    from provider.resource r join
+                    file_orchestrator.file_collections fc on r.id = fc.resource join
+                    accounting.products p on r.product = p.id join
+                    accounting.product_categories pc on p.category = pc.id
+                where 
+                    (:isproject and project = :project) or 
+                    (not :isproject and created_by = :username and project is null)
+            """.trimIndent()
+        ).rows.map { data ->
+            val title = data.getString(0)!!.lowercase()
+            val provider = data.getString(1)!!
+            val category = data.getString(2)!!
+            idWithSpec.forEach {
+                if (it.second.title.lowercase() == title && it.second.product.provider == provider && it.second.product.category == category) {
+                    return@map true
+                }
+            }
+            false
+        }.contains(true)
```

Comment:
This has the added benefit that we can terminate early instead of having to compute every single comparison.

```suggestion
            idWithSpec.any { (_, spec) ->
                spec.title.equals(title, ignoreCase = true) &&
                    spec.product.provider == provider &&
                    spec.product.category == category
            }
        }
```